plugins/modules/ios_service.py

```diff
@@ -0,0 +1,425 @@
+  cisco.ios.ios_service:
+    state: deleted
+
+# Commands Fired:
+# ---------------
+
+# "commands": [
+#        "service pad",
+#        "no service password-encryption",
+#        "no service tcp-keepalives-in",
+#        "no service tcp-keepalives-out",
+#        "no service timestamps  debug",
+#        "no service timestamps  log"
+# ]
+
+# After state:
+# ------------
+
+# switch-ios#sh run | sec ^service|^no service
+# no service timestamps debug uptime
+# no service timestamps log uptime
+
+# Using state: replaced
+
+# Before state:
+# -------------
+
+# switch-ios#sh run | sec ^service|^no service
+# no service pad
+# service tcp-keepalives-in
+# service tcp-keepalives-out
+# service timestamps debug datetime
+# service timestamps log datetime
+# service password-encryption
+
+# Delete play:
+# ------------
+
+- name: Apply the provided configuration
+  cisco.ios.ios_service:
+    config:
+      timestamps:
+        - msg: log
+          enable: true
+          timestamp: datetime
+          datetime_options:
+            localtime: true
+            msec: true
+            show_timezone: true
+            year: true
+        - msg: debug
+          enable: true
+          timestamp: datetime
+      pad: false
+      password_encryption: true
+    state: "replaced"
+
+# Commands Fired:
+# ---------------
+
+# "commands": [
+#        "no service tcp-keepalives-in",
+#        "no service tcp-keepalives-out",
+#        "no service timestamps log",
+#        "service timestamps log datetime msec localtime show-timezone year"
+# ]
+
+# After state:
+# ------------
+
+# switch-ios#sh run | sec ^service|^no service
+# no service pad
+# service timestamps debug datetime
+# service timestamps log datetime msec localtime show-timezone year
+# service password-encryption
+
+"""
+RETURN = """
+before:
+  description: The configuration as structured data prior to module invocation.
+  returned: always
+  type: dict
+  sample: The configuration returned will always be in the same format of the parameters above.
+after:
+  description: The configuration as structured data after module completion.
+  returned: when changed
+  type: dict
+  sample: The configuration returned will always be in the same format of the parameters above.
+commands:
+  description: The set of commands pushed to the remote device
+  returned: always
+  type: list
+  sample: ['lldp holdtime 10', 'lldp run', 'lldp timer 10']
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+
+from ansible_collections.cisco.ios.plugins.module_utils.network.ios.argspec.service.service import (
+    ServiceArgs,
+)
+from ansible_collections.cisco.ios.plugins.module_utils.network.ios.config.service.service import (
+    Service,
+)
+
+
+def main():
+    """
+    Main entry point for module execution
+
+    :returns: the result form module invocation
+    """
+    required_if = [
+        ("state", "merged", ("config",)),
+        ("state", "replaced", ("config",)),
```

Comment:
Do we need replaced state for this module?
Does it add any significant use case other than overridden?

Reply:
Because the `replaced` and `overridden` action have the same result, I have keep only the state `replaced`